app/services/indicator_service.js

```diff
@@ -5,23 +5,30 @@ import uuidv4 from '../utils/uuidv4';
 const saveIndicator =  (scorecardUUID, indicators, updateDownloadPercentage, callback) => {
   let savedCount = 0;
   indicators.map((indicator) => {
-    const indicatorSet = {
-      uuid: uuidv4(),
-      id: indicator.id,
-      name: indicator.name,
-      facility_id: indicator.categorizable.id,
-      scorecard_uuid: scorecardUUID,
-      tag: indicator.tag_name
-    };
-    realm.write(() => {
-      realm.create('Indicator', indicatorSet, 'modified');
-      savedCount += 1;
-    });
+    if (!_isIndicatorSaved(indicator.id)) {
+      const indicatorSet = {
+        uuid: uuidv4(),
+        id: indicator.id,
+        name: indicator.name,
+        facility_id: indicator.categorizable.id,
+        scorecard_uuid: scorecardUUID,
+        tag: indicator.tag_name
+      };
+      realm.write(() => {
+        realm.create('Indicator', indicatorSet, 'modified');
+      });
+    }
     updateDownloadPercentage(getDownloadPercentage(indicators.length));
+    savedCount += 1;
   });
   callback(savedCount === indicators.length);
 };
 
+const _isIndicatorSaved = (indicatorId) => {
```

Comment:
@limkimsan please rename this method _isIndicatorSaved(indicatorId) to isExist(indicatorId)